src/dsp_tools/models/input_error.py

```diff
@@ -0,0 +1,131 @@
+from dataclasses import dataclass
+from typing import Protocol
+
+# pylint: disable=too-few-public-methods
+
+
+separator = "\n\t"
+list_separator = "\n\t- "
+
+
+class Problem(Protocol):
+    """Information about input errors."""
+
+    def execute_error_protocol(self) -> str:
+        """
+        This function initiates all the steps for successful problem communication with the user.
+
+        Returns:
+            message for the error
+        """
+
+
+@dataclass(frozen=True)
+class DockerNotStartedProblem:
+    """This class contains information if the docker program is not started."""
+
+    user_msg: str
+
+
+@dataclass(frozen=True)
+class DirectoryProblem:
+    """This class contains information if a path does not exist."""
+
+    user_msg: str
+
+
+@dataclass(frozen=True)
+class FileNotFoundProblem:
+    """This class contains information if a file is not found at the expected location."""
+
+    user_msg: str
+
+
+@dataclass(frozen=True)
+class ExcelStructureProblem:
+    """This class contains information if the structure (column names, etc.) is incorrect."""
+
+    user_msg: str
+    column: list[str] | None = None
+
+    def execute_error_protocol(self) -> str:
+        """
+        This function initiates all the steps for successful problem communication with the user.
+
+        Returns:
+            message for the error
+        """
+        msg = self._generate_error_msg()
+        return msg
+
+    def _generate_error_msg(self) -> str:
+        msg = self.user_msg + separator
+        if self.column:
+            msg += "Column(s):" + list_separator.join(self.column)
+        return msg
+
+
+@dataclass(frozen=True)
+class ExcelContentProblem:
+    """This class contains information if the content of an Excel is not as expected."""
+
+    user_msg: str
+    column: str | None = None
+    rows: list[int] | None = None
+    values: list[str] | None = None
+
+    def execute_error_protocol(self) -> str:
+        """
+        This function initiates all the steps for successful problem communication with the user.
+
+        Returns:
+            message for the error
+        """
+        msg = self._generate_error_msg()
+        return msg
+
+    def _generate_error_msg(self) -> str:
+        msg = [self.user_msg]
+        if self.column:
+            msg.append("Column: " + self.column)
+        if self.rows:
+            rws = [str(x) for x in self.rows]
+            msg.append("Row(s):" + list_separator + list_separator.join(rws))
+        if self.values:
+            msg.append("Value(s):" + list_separator + list_separator.join(self.values))
```

Comment:
I'm having a hard time imagining how these will look, with the two list_separators... won't there be an "empty bulletpoint"?

Reply:
No, if you join a list, that has only one value, then the separator is invisible. That is the advantage of it and why I need to put a separator in before the .join as the first value does not have it

Reply:
right, makes sense